src/skottie-player.ts

```diff
@@ -554,14 +595,484 @@ export class SkottiePlayer extends LitElement {
+   */
+  private updateColors(reader: FileReader, file: File, mapKey: string): void {
+    const url = URL.createObjectURL(file);
+    const oldCanvas = document.getElementById('canvas');
+    if (oldCanvas) {
+      oldCanvas.remove();
+    }
+    const img = new Image();
+    img.onload = () => { 
+      const canvasElement = document.createElement('canvas');
+      canvasElement.setAttribute('id', 'canvas');
+      canvasElement.setAttribute('width', this.width.toString());
+      canvasElement.setAttribute('height', this.height.toString());
+      const context = canvasElement.getContext('2d');
+      if (context) {
+        context.drawImage(img, 0, 0, img.width * this.height / img.height, this.height);
+        const myData = context.getImageData(0, 0, img.width * this.height / img.height, this.height);
+        
+        const k = 5;
+        let [centers, centerCounts] = this.getDominantColors(myData, k);
+        let [index, max] = [0, 0];
+        for (let i = 0; i < centerCounts.length; i++) {
+          if (centerCounts[i] > max) {
+            max = centerCounts[i];
+            index = i;
+          }
+        }
+        const color = centers[index];
+        console.log(color);
+        console.log(centers);
+
+        this.setTemplateColors(color);
+        this.updateJson();
+        this.readAssetFile(reader, file, mapKey);
+      }
+      URL.revokeObjectURL(url);
+    }
+    img.src = url;
+  }
+
+  /**
+   * Takes in an ImageData object and finds the k most dominant colors from
+   * the image data. Also returns the number of pixels assoiated with each 
+   * color's cluster.
+   * @param myData  ImageData generated from a HTML Canvas element
+   */
+  private getDominantColors(myData: ImageData, k: number): Array<any> {
+    let dataMap, keyArray, total;
+    [dataMap, keyArray, total] = this.getDataMap(myData)
+
+    let centerCandidates = this.getCenterCandidates(dataMap, keyArray, total)
+
+    let centers = new Array<string>();
+    let centerCounts, centerToColor;
+    while (centers.length === 0) {
+      [centers, centerCounts, centerToColor] = this.kMeans(k, dataMap, centerCandidates);
+      [centers, centerCounts] = this.filterWhiteBlack(centers, centerCounts, centerToColor);
+    }
+
+    return [centers, centerCounts];
+  }
+
+  /**
+   * Gets center candidates for performing k means on the data in dataMap.
+   * Eliminates candidates if the count for a certain color is too low.
+   * 
+   * @param dataMap   Map from color to count of color in image.
+   * @param keyArray  Array containing keys of dataMap in arbitrary order.
+   * @param total     Total number of pixels from image used to make dataMap.
+   */
+  private getCenterCandidates(dataMap: Map<string, number>, 
+    keyArray: Array<string>, total: number) {
+    let centerCandidates = [];
+    for (let i = 0; i < keyArray.length; i++) {
+      const count = dataMap.get(keyArray[i])
+      if (count && total / keyArray.length / 2 < count){
```

Comment:
"total / keyArray.length / 2" could be assigned to a local variable, to avoid recompute this value in each iteration.